aten/src/ATen/native/cuda/ScaledGroupMM.cu

```diff
@@ -0,0 +1,640 @@
+    ProblemShape* problem_sizes,
+    // Strides for cutlass, cute::Stride
+    StrideA* stride_A,
+    StrideB* stride_B,
+    StrideOutput* stride_output,
+    const int32_t* offs,
+    int32_t M,
+    int32_t N,
+    int32_t K,
+    // Original strides of the input tensors
+    Strides tensor_StrideA,
+    Strides tensor_StrideB,
+    Strides tensor_StrideOutput,
+    int64_t a_scale_stride,
+    int64_t b_scale_stride) {
+  int32_t tid = threadIdx.x;
+  int32_t delta = 0;
+  if (offs != nullptr) {
+    int32_t start = tid == 0 ? 0 : offs[tid - 1];
+    delta = offs[tid] - start;
+  }
+  int64_t lda, ldb, ldoutput;
+  if (M < 0) {
+    // A and output is 2d
+    M = delta;
+    lda = tensor_StrideA.strides[0];
+    ldb = tensor_StrideB.strides[2]; // B is transposed
+    ldoutput = tensor_StrideOutput.strides[0];
+    A_ptrs[tid] = tid == 0 ? A : A + offs[tid - 1] * lda;
+    inputA_scale_ptrs[tid] = tid == 0 ? scale_A : scale_A + offs[tid - 1];
+    output_ptrs[tid] = tid == 0 ? output : output + offs[tid - 1] * ldoutput;
+    B_ptrs[tid] = B + tid * tensor_StrideB.strides[0];
+    inputB_scale_ptrs[tid] = scale_B + tid * b_scale_stride;
+  } else if (N < 0) {
+    N = delta;
+    lda = tensor_StrideA.strides[1];
+    ldb = tensor_StrideB.strides[1]; // B is transposed
+    ldoutput = tensor_StrideOutput.strides[0];
+    A_ptrs[tid] = A + tid * tensor_StrideA.strides[0];
+    inputA_scale_ptrs[tid] = scale_A + tid * a_scale_stride;
+    output_ptrs[tid] = tid == 0 ? output : output + offs[tid - 1];
+    B_ptrs[tid] = tid == 0 ? B : B + offs[tid - 1] * ldb;
+    inputB_scale_ptrs[tid] = tid == 0 ? scale_B : scale_B + offs[tid - 1];
+  } else if (K < 0) {
+    // A, B is 2d, output is 3d
+    K = delta;
+    lda = tensor_StrideA.strides[0];
+    ldb = tensor_StrideB.strides[1]; // B is transposed
+    ldoutput = tensor_StrideOutput.strides[1];
+    A_ptrs[tid] = tid == 0 ? A : A + offs[tid - 1];
+    B_ptrs[tid] = tid == 0 ? B : B + offs[tid - 1];
+    inputA_scale_ptrs[tid] = scale_A + tid * M;
+    inputB_scale_ptrs[tid] = scale_B + tid * N;
+    output_ptrs[tid] = output + tid * tensor_StrideOutput.strides[0];
+  } else {
+    // A, B, output are 3D
+    lda = tensor_StrideA.strides[1];
+    ldb = tensor_StrideB.strides[2];
+    ldoutput = tensor_StrideOutput.strides[1];
+    A_ptrs[tid] = A + tid * tensor_StrideA.strides[0];
+    B_ptrs[tid] = B + tid * tensor_StrideB.strides[0];
+    inputA_scale_ptrs[tid] = scale_A + tid * a_scale_stride;
+    inputB_scale_ptrs[tid] = scale_B + tid * b_scale_stride;
+    output_ptrs[tid] = output + tid * tensor_StrideOutput.strides[0];
+  }
+  problem_sizes[tid] = ProblemShape(M, N, K);
+
+  stride_A[tid] = cutlass::make_cute_packed_stride(StrideA{}, {M, lda, 1});
+  stride_B[tid] = cutlass::make_cute_packed_stride(StrideB{}, {N, ldb, 1});
+  stride_output[tid] =
+      cutlass::make_cute_packed_stride(StrideOutput{}, {M, ldoutput, 1});
+}
+
+constexpr int kNumSMsForH100 = 132;
```

Comment:
probably will need to obey the new sm carveout but could be follow up

Reply:
I'm not properly setting {hw_info}, I will